### PR TITLE
🧹 Remove IInProcessAnalyzer

### LINKED ISSUE #35808: Update RenameTrackingDiagnosticAnalyzer to run out-of-process
Currently `RenameTrackingDiagnosticAnalyzer` depends on data stored on `ITextBuffer` instances which is not not available out-of-process. This analyzer will cease to work in its current form when the analyzer infrastructure is moved out-of-process. The analyzer should be updated to remove the in-process dependency, or the analyzer for this feature should be replaced with an alternative implementation that does not rely on analyzers.

### PR DESCRIPTION
**Pull request by sharwell**
* Remove unnecessary uses of IInProcessAnalyzer
* ~~Ban the ongoing use of IInProcessAnalyzer~~

    RenameTrackingDiagnosticAnalyzer is the only remaining in-process analyzer, which is tracked by #35808. Use of IInProcessAnalyzer in new locations is not allowed.

* After changing `RenameTrackingDiagnosticAnalyzer` to extend `DocumentDiagnosticAnalyzer`, the `IInProcessAnalyzer` interface is removed entirely

    Closes #35808

### CODE REVIEW

File: src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 namespace Microsoft.CodeAnalysis.RemoveUnnecessaryImports
 {
     internal abstract class AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer
-        : DiagnosticAnalyzer, IBuiltInAnalyzer, IInProcessAnalyzer
+        : DiagnosticAnalyzer, IBuiltInAnalyzer
```
Comment (heejaechang) [inline src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs]: so, do we know whether these analyzers can actually run in OOP?

Comment (heejaechang) [reply src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs]: if these analyzer now can run in OOP, then that would be awesome.

Comment (sharwell) [reply src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs]: I looked through the services it depends on, and they all appear to be defined in the features layer, which would make them available OOP.

**Comment by sharwell** [comment]
@heejaechang Can you look at the last two commits and let me know if that's what you had in mind?

File: src/EditorFeatures/Core/Implementation/RenameTracking/AbstractRenameTrackingCodeFixProvider.cs [hunk 2]
```diff
@@ -29,7 +29,9 @@ internal abstract class AbstractRenameTrackingCodeFixProvider : CodeFixProvider
 
         public sealed override ImmutableArray<string> FixableDiagnosticIds
         {
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/RenameTracking/AbstractRenameTrackingCodeFixProvider.cs]: what is this warning about? I don't see any obsolete?

File: src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.cs [hunk 3]
```diff
@@ -144,8 +145,10 @@ internal static async Task<IEnumerable<Diagnostic>> GetDiagnosticsAsync(SyntaxTr
 
             var message = string.Format(
                 EditorFeaturesResources.Rename_0_to_1,
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.cs]: not sure what are obsolete.

File: src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs [hunk 4]
```diff
@@ -176,7 +176,9 @@ public async Task AssertNoTag()
         public async Task<IList<Diagnostic>> GetDocumentDiagnosticsAsync(Document document = null)
         {
             document = document ?? this.Workspace.CurrentSolution.GetDocument(_hostDocument.Id);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (heejaechang) [inline src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs]: same here?

Comment (mavasani) [reply src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs]: Yeah all pragmas should now be reverted given RenameTracking analyzer is now a document diagnostic analyzer.

File: src/EditorFeatures/Core/Implementation/RenameTracking/AbstractRenameTrackingCodeFixProvider.cs (hunk 2 above)
Comment (sharwell) [reply src/EditorFeatures/Core/Implementation/RenameTracking/AbstractRenameTrackingCodeFixProvider.cs]: Thanks, I forgot to remove these lines when I removed `[Obsolete]` from `RenameTrackingDiagnosticAnalyzer`.

File: src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.cs (hunk 3 above)
Comment (sharwell) [reply src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.cs]: Thanks, I forgot to remove these lines when I removed `[Obsolete]` from `RenameTrackingDiagnosticAnalyzer`.

File: src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs (hunk 4 above)
Comment (sharwell) [reply src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs]: Thanks, I forgot to remove these lines when I removed `[Obsolete]` from `RenameTrackingDiagnosticAnalyzer`.

**Pull request merged**